### PR TITLE
add fcmOptions in WebpushConfig

### PR DESCRIPTION
**Pull request by chemidy**
new option in webpushconfig 
#182

### CODE REVIEW

**Comment by chemidy** [comment]
Formatting has changed on [go 1.11](https://golang.org/doc/go1.11#gofmt) check the reason here : https://github.com/golang/go/commit/542ea5ad91367d2b40589942cc5757a4d5f43f97

So formating has changed 
-		"to": "/topics/test-topic",
+		"to":                  "/topics/test-topic",

this is why the build failed

**Comment by chemidy** [comment]
Maybe we can just gofmt ( ./.travis.gofmt.sh) for go >= 1.11 due to incompatible changes

**Comment by hiranya911** [comment]
We currently don't test on go 1.11 so I think we can ignore the gofmt issue for now.

File: .travis.yml [hunk 1]
```diff
@@ -20,14 +20,14 @@ go_import_path: firebase.google.com/go
 
 before_install:
     # Golint requires Go 1.7 or later.
-    - if ! [[ "$TRAVIS_GO_VERSION" =~ ^1\.6\.([0-9]+|x)$ ]]; then go get github.com/golang/lint/golint; fi
+    - if ! [[ "$TRAVIS_GO_VERSION" =~ ^1\.6\.([0-9]+|x)$ || "$TRAVIS_GO_VERSION" =~ ^1\.7\.([0-9]+|x)$ || "$TRAVIS_GO_VERSION" =~ ^1\.8\.([0-9]+|x)$ ]]; then go get golang.org/x/lint/golint; fi
```
Comment (hiranya911) [inline .travis.yml]: cloud.google.com/go has stopped supporting go 1.8 and earlier, which means we can stop supporting them too. Can you drop those targets from the Travis config, and update this file accordingly?

File: messaging/messaging.go [hunk 2]
```diff
@@ -225,6 +224,7 @@ type WebpushConfig struct {
 	Headers      map[string]string    `json:"headers,omitempty"`
 	Data         map[string]string    `json:"data,omitempty"`
 	Notification *WebpushNotification `json:"notification,omitempty"`
+	FcmOptions   *WebpushFcmOptions   `json:"fcm_options,omitempty"`
```
Comment (hiranya911) [inline messaging/messaging.go]: FCM team tells me that this json key should be called `fcmOptions`. `fcm_options` is a documentation error, and they are working on it.

Comment (chemidy) [reply messaging/messaging.go]: OK :) fixed

File: .travis.yml (hunk 1 above)
Comment (chemidy) [reply .travis.yml]: OK, now that we have removed support 1.6. Maybe we can update context Import `golang.org/x/netcontext` -> `context` and other tricks for 1.6 in another PR.

Comment (hiranya911) [reply .travis.yml]: Yes, lets make those changes in a separate PR.

**Pull request merged**